### PR TITLE
dom/met signature: `make_base_discrete_gaussian`

### PR DESCRIPTION
**Pull request by Shoeboxam**

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #696** <a href="https://app.graphite.dev/github/pr/opendp/opendp/696" target="_blank">Graphite</a> 
    * **PR #645** <a href="https://app.graphite.dev/github/pr/opendp/opendp/645" target="_blank">Graphite</a> 
      * **PR #691** <a href="https://app.graphite.dev/github/pr/opendp/opendp/691" target="_blank">Graphite</a> 
      * **PR #604** <a href="https://app.graphite.dev/github/pr/opendp/opendp/604" target="_blank">Graphite</a> 
        * **PR #706** <a href="https://app.graphite.dev/github/pr/opendp/opendp/706" target="_blank">Graphite</a> 
        * **PR #699** <a href="https://app.graphite.dev/github/pr/opendp/opendp/699" target="_blank">Graphite</a> 
        * **PR #687** <a href="https://app.graphite.dev/github/pr/opendp/opendp/687" target="_blank">Graphite</a> 
          * **PR #688** <a href="https://app.graphite.dev/github/pr/opendp/opendp/688" target="_blank">Graphite</a> 
            * **PR #689** <a href="https://app.graphite.dev/github/pr/opendp/opendp/689" target="_blank">Graphite</a> 
              * **PR #749** <a href="https://app.graphite.dev/github/pr/opendp/opendp/749" target="_blank">Graphite</a> 
              * **PR #712** <a href="https://app.graphite.dev/github/pr/opendp/opendp/712" target="_blank">Graphite</a> 
                * **PR #713** <a href="https://app.graphite.dev/github/pr/opendp/opendp/713" target="_blank">Graphite</a> 
                  * **PR #736** <a href="https://app.graphite.dev/github/pr/opendp/opendp/736" target="_blank">Graphite</a> 
                    * **PR #761** <a href="https://app.graphite.dev/github/pr/opendp/opendp/761" target="_blank">Graphite</a> 
                      * **PR #798** <a href="https://app.graphite.dev/github/pr/opendp/opendp/798" target="_blank">Graphite</a> 
                        * **PR #799** <a href="https://app.graphite.dev/github/pr/opendp/opendp/799" target="_blank">Graphite</a>  👈
                      * **PR #796** <a href="https://app.graphite.dev/github/pr/opendp/opendp/796" target="_blank">Graphite</a> 
              * **PR #690** <a href="https://app.graphite.dev/github/pr/opendp/opendp/690" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/799?utm_source=stack-comment).

File: docs/source/user/measurements/additive-noise-mechanisms.ipynb [hunk 1]
```diff
@@ -397,8 +398,9 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Regardless of the choice of `D`, the discrete versions of the measurements still expect integral sensitivities.\n",
-    "A special affordance is made for the vector-valued discrete Gaussian mechanism, which can be configured to expect a float sensitivity instead:"
+    "The vector-valued discrete Gaussian mechanism has a special affordance \n",
+    "in that the input metric can be configured to expect a float instead.\n",
+    "This is because there is often a square root in sensitivity calculations for the "
```
Comment (andrewvyrros) [inline docs/source/user/measurements/additive-noise-mechanisms.ipynb:403]: Unfinished sentence.

File: rust/src/measurements/discrete_gaussian/mod.rs [hunk 2]
```diff
@@ -90,6 +90,8 @@ where
 /// | `VectorDomain<AtomDomain<T>>` | `Vec<T>`     | `L2Distance<QI>`        |
 ///
 /// # Arguments
+/// * `input_domain` - Domain of the data type to be privatized. Valid values are `VectorDomain<AtomDomain<T>>` or `AtomDomain<T>`.
+/// * `input_metric` - Metric of the data type to be privatized. Valid values are `AbsoluteDistance<T>` or `L2Distance<T>`.
```
Comment (andrewvyrros) [inline rust/src/measurements/discrete_gaussian/mod.rs:94]: I thought you were switching these to use constructor form (i.e. `atom_domain(T)`)?

File: rust/src/measurements/gaussian/mod.rs [hunk 3]
```diff
@@ -79,13 +79,12 @@ where
 /// If k is not set, k defaults to the smallest granularity.
 ///
 /// # Arguments
-/// * `input_domain` - Domain of the data type to be privatized.
-/// * `input_metric` - Metric of the data type to be privatized.
+/// * `input_domain` - Domain of the data type to be privatized. Valid values are `VectorDomain<AtomDomain<T>>` or `AtomDomain<T>`.
+/// * `input_metric` - Metric of the data type to be privatized. Valid values are `AbsoluteDistance<T>` or `L2Distance<T>`.
```
Comment (andrewvyrros) [inline rust/src/measurements/gaussian/mod.rs:83]: I thought you were switching these to use constructor form (i.e. `atom_domain(T)`)?

File: docs/source/user/measurements/additive-noise-mechanisms.ipynb (hunk 1 above)
Comment (Shoeboxam) [reply docs/source/user/measurements/additive-noise-mechanisms.ipynb:403]: Thanks, I got sidetracked there.

File: rust/src/measurements/discrete_gaussian/mod.rs (hunk 2 above)
Comment (Shoeboxam) [reply rust/src/measurements/discrete_gaussian/mod.rs]: Dropped the second sentence from each, and updated the table. Also brought back the doc comments for the generics. The suppressed generic docs don't appear in Python.

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**